sdks/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,241 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required (VERSION 3.13.0)
+
+project(agones VERSION 0.8.0 HOMEPAGE_URL https://github.com/GoogleCloudPlatform/agones LANGUAGES C CXX)
+
+# Build options
+option(AGONES_BUILD_SHARED "Build Agones C++ SDK as dynamic library" ON)
+option(AGONES_FORCE_GRPC_VERSION "Build Agones C++ SDK only with officially supported gRPC version" ON)
+option(AGONES_CREATE_PACKAGE "Generate CMake installation step and package files for Agones C++ SDK" OFF)
```

Comment:
I just tried with this enabled, and I got an error:
` cmake .. -G "Visual Studio 15 2017 Win64" -Wno-dev -DAGONES_BUILD_SHARED=OFF -DAGONES_CREATE_PACKAGE=ON`
```
-- Performing Test HAVE_STEADY_CLOCK -- success
CMake Error: Could not open file for write in copy operation C:/Program Files/agones/cmake/agonesConfig.cmake.tmp
CMake Error: : System Error: No such file or directory
CMake Error at C:/Program Files/CMake/share/cmake-3.13/Modules/CMakePackageConfigHelpers.cmake:316 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  CMakeLists.txt:189 (configure_package_config_file)
```

Reply:
Good catch. I'll fix it. I need to store package config to the output folder first. 
In your case, even if build will be successful, you can't do an install step, because your compiler has no write access to default `CMAKE_INSTALL_PREFIX` path (which is Program Files).